scripts/build.sh

```diff
@@ -108,8 +108,17 @@ main() {
   fi
 
   if [[ $CROSS_ARCH != "" ]]; then
-    if [[ $platform == "macos" && $CROSS_ARCH == "arm64" ]]; then
+    if [[ $platform == "macos" ]]; then
+      macos_version_min=10.11
+      if [[ $CROSS_ARCH == "arm64" ]]; then
         cross_file="--cross-file resources/macos/macos_arm64.conf"
+        macos_version_min=11.0
+      fi
+      export MACOSX_DEPLOYMENT_TARGET=$macos_version_min
```

Comment:
this line is required to force the compiler to use the older Objective-C ABI, or else there will be `Symbol not found: _objc_alloc_init` on macOS 10.11 ~ 10.13